std-bits/table/src/main/java/org/enso/table/data/column/builder/object/InferredBuilder.java

```diff
@@ -87,39 +61,60 @@ public void append(Object o) {
       currentBuilder.append(o);
     } else {
       switch (currentBuilder.getType()) {
-        case Storage.Type.BOOL:
+        case Storage.Type.BOOL -> {
           if (o instanceof Boolean) {
             currentBuilder.append(o);
```

Comment:
Could we make `append` return True/False and make the currentBuilder make this decision.

Reply:
the `accepts` method already answers this so feel like we can do better here.

Reply:
right!